src/Renci.SshNet/ScpClient.cs

```diff
@@ -579,6 +583,11 @@ private void UploadFileContent(IChannelSession channel, Stream input, Stream sou
                 read = source.Read(buffer, 0, buffer.Length);
             }
 
+            if (totalLength == 0 && totalRead == 0 && notifyOnEmptyFile)
+            {
+                RaiseUploadingEvent(remoteFileName, totalLength, totalRead);
+            }
```

Comment:
Rather than changing the API, I would be happy just to take these 3 lines

```suggestion
            if (totalLength == 0 && totalRead == 0)
            {
                RaiseUploadingEvent(remoteFileName, totalLength, totalRead);
            }
```

Reply:
Updated!